code/modules/GOAP/cyka_blyat/actions.dm

```diff
@@ -0,0 +1,339 @@
+	if(A.grenade_to_throw == null || A.grenades_left < 1)
+		return FALSE
+	var/list/viewl = view(10, A)
+	for(var/mob/living/carbon/C in viewl)
+		if(C && !C.stat && !A.CheckFriendlyFire(C))
+			target = C
+	return (target != null)
+
+/datum/goap_action/russian/grenade_out_take_cover/RequiresInRange()
+	return FALSE
+
+/datum/goap_action/russian/grenade_out_take_cover/Perform(mob/living/simple_animal/hostile/russian/ranged/A)
+	cooldown_time = world.time + cooldown
+	var/mob/living/carbon/C = target
+	A.say(pick(list("GRENADA VYKHODA KRYSHKA!","GRANADA, GRANADA!","LOZHIS'!")))
+	A.visible_message("<span class = 'danger'>[A] throws a grenade at [C]!</span>")
+	var/obj/item/grenade/G = new A.grenade_to_throw(get_turf(A))
+	G.preprime(A, 5)
+	G.throw_at(C, 10, 1)
+	A.grenades_left--
+	action_done = TRUE
+	return TRUE
+
+/datum/goap_action/russian/grenade_out_take_cover/CheckDone(atom/agent)
+	return action_done
+
+
+/datum/goap_action/russian/reload
+	name = "Reload"
+	cost = 1 // reload + firing = 3, so this comes if you can't fire/reload+fire/
+
+/datum/goap_action/russian/reload/New()
+	..()
+	preconditions = list()
+	preconditions["hasAmmo"] = FALSE
+	effects = list()
+	effects["hasAmmo"] = TRUE
+
+/datum/goap_action/russian/reload/AdvancedPreconditions(mob/living/simple_animal/hostile/russian/ranged/A, list/worldstate)
+	if(A.reloads_left)
+		return TRUE
+	else
+		return FALSE
+
+/datum/goap_action/russian/reload/RequiresInRange()
+	return FALSE
+
+/datum/goap_action/russian/reload/Perform(mob/living/simple_animal/hostile/russian/ranged/A)
+	A.say(pick(list("REKLAMA, KRYSHKA MENYA!","Novyy mag, day mne sekundu!","Peregruzochnyy!")))
+	A.ammo_left = A.max_ammo
+	A.reloads_left--
+	action_done = TRUE
+	return TRUE
+
+/datum/goap_action/russian/reload/CheckDone(atom/agent)
+	return action_done
+
+
+/datum/goap_action/russian/medic
+	name = "Heal Allies"
+	cost = 1
+	cooldown = 10
+
+/datum/goap_action/russian/medic/New()
+	..()
+	preconditions = list()
+	preconditions["allyHealed"] = FALSE
+	effects = list()
+	effects["allyHealed"] = TRUE
+
+/datum/goap_action/russian/medic/AdvancedPreconditions(atom/agent, list/worldstate)
+	var/list/viewl = view(10, agent)
+	for(var/mob/living/simple_animal/hostile/russian/ranged/R in viewl)
+		if(!R.stat)
+			if(R.health < (R.maxHealth/2))
+				target = R
+				break
+	return (target != null)
+
+/datum/goap_action/russian/medic/RequiresInRange()
+	return TRUE
+
+/datum/goap_action/russian/medic/Perform(mob/living/simple_animal/hostile/russian/ranged/A)
+	if(target)
+		A.say(pick(list("Isprav'te sebya!","Medik na stsene!","Ostanovite poluchat' vystrel tak mnogo!")))
+		if(prob(1))
+			A.visible_message("[A] hands [target] some vodka! [target] chugs it, feeling rejuvenated!")
+		else
+			A.visible_message(pick(list("[A] applies some QuikClot to [target]!", "[A] bandages some wounds on [target]!", "[A] quickly stitches close a cut on [target]!")))
```

Comment:
This should be one of our currently existing chems instead of an American brand name